runtime/src/chpl-qsbr.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2004-2018 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chplrt.h"
+#include "chpl-thread-local-storage.h"
+#include "chpl-mem.h"
+#include "chpl-atomics.h"
+#include "chpl-tasks.h"
+#include "chpl-qsbr.h"
+
+static uint64_t global_epoch = 0;
```

Comment:
Does this need to be atomic?

Reply:
I use Chapel's `atomic_*` operations on it. Oh wait, or does that not work?

Reply:
That should work - I was just asking you to label the type as 'atomic'